### ISSUE TITLE
Webdav COPY operation fails from one shared folder to another

### ISSUE THREAD

**Issue by davidgubler** [nextcloud/server#6071]
### Steps to reproduce
1. Create two directories (a and b) in the admin account
2. Create regular user "foo"
3. Share both directories with user "foo"
4. Open Webdav in Nautilus (Ubuntu 16.04): Connect to server, use URL dav(s)://server/remote.php/webdav/ (see NOTE 2 below), log in using the "foo" user
5. create a file in directory a, right-click copy, go to directory b, right-click paste

### Expected behaviour
File gets copied from directory a to directory b

### Actual behaviour
Nextcloud throws error:
{"reqId":"uQEh1lO0GWU4sw1ipmit","level":3,"time":"2017-08-10T12:49:12+00:00","remoteAddr":"192.168.216.1","user":"foo","app":"PHP","method":"COPY","url":"\/remote.php\/webdav\/sys\/smsBox-v4.2-HTTP-XML-API.pdf","message":"copy(\/\/var\/www\/barcloud\/public_html\/data\/admin\/smsBox-v4.2-HTTP-XML-API.pdf): failed to open stream: No such file or directory at \/opt\/nextcloud-12.0.1\/nextcloud\/lib\/private\/Files\/Storage\/Local.php#272","userAgent":"gvfs\/1.28.2","version":"12.0.1.5"}

NOTE 1: Even with the superfluous slash at the beginning of the path removed, the path is wrong. It misses the "files" directory (part of nextcloud's data layout) and the "sys" directory (shared directory):
/var/www/barcloud/public_html/data/admin/files/sys/smsBox-v4.2-HTTP-XML-API.pdf actually exists.

NOTE 2: If I use the URL dav(s)://server/remote.php/dav/files/USERNAME/, the same operation works fine (!). Looking at the code it seems that the two URLs use completely different code paths, but somehow they both implement Webdav?

NOTE 3: The web interface suggests using the /remote.php/webdav/ URL, even though according to the version numbers in the "dav" app this seems to be the older version. Older documentation suggests using the /remote.php/dav/files/USERNAME URL, see https://docs.nextcloud.com/server/9/user_manual/files/access_webdav.html

### Server configuration

**Operating system**: Ubuntu 16.04

**Web server:** nginx

**Database:** MySQL 5.7.17

**PHP version:** 7.0 through PHP-FPM

**Nextcloud version:** 12.0.1 (same error on 12.0.0)

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh install

**Where did you install Nextcloud from:** .tar.gz downloaded from https://nextcloud.com/

**Signing status:**
<details>
<summary>Signing status</summary>
```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.5.2
  - bruteforcesettings: 1.0.2
  - comments: 1.2.0
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - twofactor_totp: 1.3.0
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "datadirectory": "\/var\/www\/barcloud\/public_html\/data",
        "appstoreenabled": false,
        "instanceid": "ocvpe05voiq7",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "barcloud.vagrant.dev",
            "192.168.216.49"
        ],
        "overwrite.cli.url": "https:\/\/barcloud.vagrant.dev",
        "dbtype": "mysql",
        "version": "12.0.1.5",
        "dbname": "barcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "nc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 2,
        "maintenance": false
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
Nautilus 3.14.3, stock Ubuntu 16.04 file browser

#### Web server error log
none

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"uQEh1lO0GWU4sw1ipmit","level":3,"time":"2017-08-10T12:49:12+00:00","remoteAddr":"192.168.216.1","user":"foo","app":"PHP","method":"COPY","url":"\/remote.php\/webdav\/sys\/smsBox-v4.2-HTTP-XML-API.pdf","message":"copy(\/\/var\/www\/barcloud\/public_html\/data\/admin\/smsBox-v4.2-HTTP-XML-API.pdf): failed to open stream: No such file or directory at \/opt\/nextcloud-12.0.1\/nextcloud\/lib\/private\/Files\/Storage\/Local.php#272","userAgent":"gvfs\/1.28.2","version":"12.0.1.5"}
```
</details>

#### Browser log
not applicable

**Comment by MorrisJobke** [comment]
cc @rullzer @icewind1991 @georgehrke

**Comment by rullzer** [comment]
Please check if this is still the case on 12.0.2. I think we fixed it there.

**Comment by MorrisJobke** [comment]
Then let's close it and if it still a problem with the latest version, we can reopen, okay? :)

**Comment by davidgubler** [comment]
Our customer confirms that this problem is fixed with 12.0.2. Thank you very much!
